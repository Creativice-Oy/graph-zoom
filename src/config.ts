import {
  IntegrationExecutionContext,
  IntegrationValidationError,
  IntegrationInstanceConfigFieldMap,
  IntegrationInstanceConfig,
} from '@jupiterone/integration-sdk-core';
import { createAPIClient } from './client';

export const instanceConfigFields: IntegrationInstanceConfigFieldMap = {
  zoomAccessToken: {
    type: 'string',
    mask: true,
  },
  scopes: {
    type: 'string',
  },
};

export interface IntegrationConfig extends IntegrationInstanceConfig {
  /**
   * The OAuth token used to authenticate requests.
   */
  zoomAccessToken: string;
  scopes: string;
}

export async function validateInvocation(
  context: IntegrationExecutionContext<IntegrationConfig>,
) {
  const { config } = context.instance;

  if (!config.zoomAccessToken || !config.scopes) {
    throw new IntegrationValidationError(
      'Config requires zoom access token and scopes field. Please generate from OAuth server.',
    );
  }

  const apiClient = createAPIClient(config);
  await apiClient.verifyAuthentication();
}
