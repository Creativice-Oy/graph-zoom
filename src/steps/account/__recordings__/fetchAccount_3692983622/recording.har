{
  "log": {
    "_recordingName": "fetchAccount",
    "creator": {
      "comment": "persister:JupiterOneIntegationFSPersister",
      "name": "Polly.JS",
      "version": "4.3.0"
    },
    "entries": [
      {
        "_id": "32b6c540dbe7339d81f7948939897f76",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "*/*"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip,deflate"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "api.zoom.us"
            }
          ],
          "headersSize": 862,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [],
          "url": "https://api.zoom.us/v2/users/me"
        },
        "response": {
          "bodySize": 739,
          "content": {
            "mimeType": "application/json;charset=UTF-8",
            "size": 739,
            "text": "{\"id\":\"H9Yx0-O8QUCXVIseYAsjaA\",\"first_name\":\"JupiterOne\",\"last_name\":\"Development\",\"email\":\"development@jupiterone.dev\",\"type\":2,\"role_name\":\"Owner\",\"pmi\":4440405982,\"use_pmi\":false,\"personal_meeting_url\":\"https://zoom.us/j/4440405982?pwd=c0Jidml0ZS9ZRzA3dEZWSTNHNHlPZz09\",\"timezone\":\"America/New_York\",\"verified\":1,\"dept\":\"\",\"created_at\":\"2021-10-27T00:00:49Z\",\"last_login_time\":\"2021-11-02T09:51:57Z\",\"host_key\":\"815201\",\"cms_user_id\":\"\",\"jid\":\"h9yx0-o8qucxviseyasjaa@xmpp.zoom.us\",\"group_ids\":[\"ppUAdwEBRFCdRE0mfLttog\"],\"im_group_ids\":[],\"account_id\":\"N0oChUdLTrCZLfk9JXp3Hw\",\"language\":\"\",\"phone_country\":\"\",\"phone_number\":\"\",\"status\":\"active\",\"job_title\":\"\",\"location\":\"\",\"login_types\":[100],\"role_id\":\"0\",\"account_number\":7011249145}"
          },
          "cookies": [
            {
              "domain": ".zoom.us",
              "expires": "1970-01-01T00:00:10.000Z",
              "httpOnly": true,
              "name": "zm_aid",
              "path": "/",
              "secure": true,
              "value": "[REDACTED]"
            },
            {
              "domain": ".zoom.us",
              "expires": "1970-01-01T00:00:10.000Z",
              "httpOnly": true,
              "name": "zm_haid",
              "path": "/",
              "secure": true,
              "value": "[REDACTED]"
            },
            {
              "domain": ".zoom.us",
              "expires": "1970-01-01T00:00:10.000Z",
              "httpOnly": true,
              "name": "zm_tmaid",
              "path": "/",
              "secure": true,
              "value": "[REDACTED]"
            },
            {
              "domain": ".zoom.us",
              "expires": "1970-01-01T00:00:10.000Z",
              "httpOnly": true,
              "name": "zm_htmaid",
              "path": "/",
              "secure": true,
              "value": "[REDACTED]"
            },
            {
              "httpOnly": true,
              "name": "cred",
              "path": "/",
              "secure": true,
              "value": "[REDACTED]"
            },
            {
              "domain": ".zoom.us",
              "expires": "1970-01-01T00:00:10.000Z",
              "httpOnly": true,
              "name": "_zm_ctaid",
              "path": "/",
              "secure": true,
              "value": "[REDACTED]"
            },
            {
              "domain": ".zoom.us",
              "expires": "1970-01-01T00:00:10.000Z",
              "httpOnly": true,
              "name": "_zm_chtaid",
              "path": "/",
              "secure": true,
              "value": "[REDACTED]"
            }
          ],
          "headers": [
            {
              "name": "date",
              "value": "Wed, 03 Nov 2021 04:43:44 GMT"
            },
            {
              "name": "content-type",
              "value": "application/json;charset=UTF-8"
            },
            {
              "name": "content-length",
              "value": "739"
            },
            {
              "name": "connection",
              "value": "close"
            },
            {
              "name": "x-zm-trackingid",
              "value": "v=2.0;clid=aw1;rid=WEB_6b57aea1eaef35fe8e753d8acb99267a"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "cache-control",
              "value": "no-cache, no-store, must-revalidate, no-transform"
            },
            {
              "name": "pragma",
              "value": "no-cache"
            },
            {
              "name": "expires",
              "value": "Thu, 01 Jan 1970 00:00:00 GMT"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "name": "vary",
              "value": "Origin, Access-Control-Request-Method, Access-Control-Request-Headers"
            },
            {
              "name": "x-ratelimit-category",
              "value": "Light"
            },
            {
              "name": "x-zm-zoneid",
              "value": "VA2"
            },
            {
              "name": "strict-transport-security",
              "value": "max-age=31536000; includeSubDomains"
            },
            {
              "name": "x-xss-protection",
              "value": "1; mode=block"
            },
            {
              "name": "referrer-policy",
              "value": "strict-origin-when-cross-origin"
            }
          ],
          "headersSize": 1347,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2021-11-03T04:43:43.081Z",
        "time": 1176,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 1176
        }
      }
    ],
    "pages": [],
    "version": "1.2"
  }
}
