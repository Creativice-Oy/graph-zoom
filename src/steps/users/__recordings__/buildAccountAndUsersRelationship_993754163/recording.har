{
  "log": {
    "_recordingName": "buildAccountAndUsersRelationship",
    "creator": {
      "comment": "persister:JupiterOneIntegationFSPersister",
      "name": "Polly.JS",
      "version": "4.3.0"
    },
    "entries": [
      {
        "_id": "32b6c540dbe7339d81f7948939897f76",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "*/*"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip,deflate"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "api.zoom.us"
            }
          ],
          "headersSize": 862,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [],
          "url": "https://api.zoom.us/v2/users/me"
        },
        "response": {
          "bodySize": 739,
          "content": {
            "mimeType": "application/json;charset=UTF-8",
            "size": 739,
            "text": "{\"id\":\"H9Yx0-O8QUCXVIseYAsjaA\",\"first_name\":\"JupiterOne\",\"last_name\":\"Development\",\"email\":\"development@jupiterone.dev\",\"type\":2,\"role_name\":\"Owner\",\"pmi\":4440405982,\"use_pmi\":false,\"personal_meeting_url\":\"https://zoom.us/j/4440405982?pwd=c0Jidml0ZS9ZRzA3dEZWSTNHNHlPZz09\",\"timezone\":\"America/New_York\",\"verified\":1,\"dept\":\"\",\"created_at\":\"2021-10-27T00:00:49Z\",\"last_login_time\":\"2021-11-02T09:51:57Z\",\"host_key\":\"815201\",\"cms_user_id\":\"\",\"jid\":\"h9yx0-o8qucxviseyasjaa@xmpp.zoom.us\",\"group_ids\":[\"ppUAdwEBRFCdRE0mfLttog\"],\"im_group_ids\":[],\"account_id\":\"N0oChUdLTrCZLfk9JXp3Hw\",\"language\":\"\",\"phone_country\":\"\",\"phone_number\":\"\",\"status\":\"active\",\"job_title\":\"\",\"location\":\"\",\"login_types\":[100],\"role_id\":\"0\",\"account_number\":7011249145}"
          },
          "cookies": [
            {
              "domain": ".zoom.us",
              "expires": "1970-01-01T00:00:10.000Z",
              "httpOnly": true,
              "name": "zm_aid",
              "path": "/",
              "secure": true,
              "value": "[REDACTED]"
            },
            {
              "domain": ".zoom.us",
              "expires": "1970-01-01T00:00:10.000Z",
              "httpOnly": true,
              "name": "zm_haid",
              "path": "/",
              "secure": true,
              "value": "[REDACTED]"
            },
            {
              "domain": ".zoom.us",
              "expires": "1970-01-01T00:00:10.000Z",
              "httpOnly": true,
              "name": "zm_tmaid",
              "path": "/",
              "secure": true,
              "value": "[REDACTED]"
            },
            {
              "domain": ".zoom.us",
              "expires": "1970-01-01T00:00:10.000Z",
              "httpOnly": true,
              "name": "zm_htmaid",
              "path": "/",
              "secure": true,
              "value": "[REDACTED]"
            },
            {
              "httpOnly": true,
              "name": "cred",
              "path": "/",
              "secure": true,
              "value": "[REDACTED]"
            },
            {
              "domain": ".zoom.us",
              "expires": "1970-01-01T00:00:10.000Z",
              "httpOnly": true,
              "name": "_zm_ctaid",
              "path": "/",
              "secure": true,
              "value": "[REDACTED]"
            },
            {
              "domain": ".zoom.us",
              "expires": "1970-01-01T00:00:10.000Z",
              "httpOnly": true,
              "name": "_zm_chtaid",
              "path": "/",
              "secure": true,
              "value": "[REDACTED]"
            }
          ],
          "headers": [
            {
              "name": "date",
              "value": "Wed, 03 Nov 2021 05:11:52 GMT"
            },
            {
              "name": "content-type",
              "value": "application/json;charset=UTF-8"
            },
            {
              "name": "content-length",
              "value": "739"
            },
            {
              "name": "connection",
              "value": "close"
            },
            {
              "name": "x-zm-trackingid",
              "value": "v=2.0;clid=aw1;rid=WEB_acedaf52eab9053e1ec7be1e033acf27"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "cache-control",
              "value": "no-cache, no-store, must-revalidate, no-transform"
            },
            {
              "name": "pragma",
              "value": "no-cache"
            },
            {
              "name": "expires",
              "value": "Thu, 01 Jan 1970 00:00:00 GMT"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "name": "vary",
              "value": "Origin, Access-Control-Request-Method, Access-Control-Request-Headers"
            },
            {
              "name": "x-ratelimit-category",
              "value": "Light"
            },
            {
              "name": "x-zm-zoneid",
              "value": "VA2"
            },
            {
              "name": "strict-transport-security",
              "value": "max-age=31536000; includeSubDomains"
            },
            {
              "name": "x-xss-protection",
              "value": "1; mode=block"
            },
            {
              "name": "referrer-policy",
              "value": "strict-origin-when-cross-origin"
            }
          ],
          "headersSize": 1347,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2021-11-03T05:11:51.725Z",
        "time": 1330,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 1330
        }
      },
      {
        "_id": "7732720bbb1de5f1637ea75eb65eddeb",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "*/*"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip,deflate"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "api.zoom.us"
            }
          ],
          "headersSize": 886,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [
            {
              "name": "page_size",
              "value": "30"
            },
            {
              "name": "page_number",
              "value": "1"
            }
          ],
          "url": "https://api.zoom.us/v2/users?page_size=30&page_number=1"
        },
        "response": {
          "bodySize": 895,
          "content": {
            "mimeType": "application/json;charset=UTF-8",
            "size": 895,
            "text": "{\"page_count\":1,\"page_number\":1,\"page_size\":30,\"total_records\":2,\"users\":[{\"id\":\"a18JbG_DQketjPo0Y8wYqg\",\"first_name\":\"J1\",\"last_name\":\"Test\",\"email\":\"tj2488801@gmail.com\",\"type\":1,\"pmi\":2434936199,\"timezone\":\"\",\"verified\":1,\"dept\":\"Test\",\"created_at\":\"2021-10-28T03:49:40Z\",\"last_login_time\":\"2021-10-28T03:52:40Z\",\"pic_url\":\"https://lh3.googleusercontent.com/a/AATXAJy4ZWS6WyBZvx4Ujm476PKw65aih3KL0KVnJBTS=s96-c\",\"group_ids\":[\"ppUAdwEBRFCdRE0mfLttog\"],\"language\":\"en-US\",\"phone_number\":\"\",\"status\":\"active\",\"role_id\":\"2\"},{\"id\":\"H9Yx0-O8QUCXVIseYAsjaA\",\"first_name\":\"JupiterOne\",\"last_name\":\"Development\",\"email\":\"development@jupiterone.dev\",\"type\":2,\"pmi\":4440405982,\"timezone\":\"America/New_York\",\"verified\":1,\"created_at\":\"2021-10-27T00:00:49Z\",\"last_login_time\":\"2021-11-02T09:51:57Z\",\"group_ids\":[\"ppUAdwEBRFCdRE0mfLttog\"],\"language\":\"\",\"phone_number\":\"\",\"status\":\"active\",\"role_id\":\"0\"}]}"
          },
          "cookies": [
            {
              "domain": ".zoom.us",
              "expires": "1970-01-01T00:00:10.000Z",
              "httpOnly": true,
              "name": "zm_aid",
              "path": "/",
              "secure": true,
              "value": "[REDACTED]"
            },
            {
              "domain": ".zoom.us",
              "expires": "1970-01-01T00:00:10.000Z",
              "httpOnly": true,
              "name": "zm_haid",
              "path": "/",
              "secure": true,
              "value": "[REDACTED]"
            },
            {
              "domain": ".zoom.us",
              "expires": "1970-01-01T00:00:10.000Z",
              "httpOnly": true,
              "name": "zm_tmaid",
              "path": "/",
              "secure": true,
              "value": "[REDACTED]"
            },
            {
              "domain": ".zoom.us",
              "expires": "1970-01-01T00:00:10.000Z",
              "httpOnly": true,
              "name": "zm_htmaid",
              "path": "/",
              "secure": true,
              "value": "[REDACTED]"
            },
            {
              "httpOnly": true,
              "name": "cred",
              "path": "/",
              "secure": true,
              "value": "[REDACTED]"
            },
            {
              "domain": ".zoom.us",
              "expires": "1970-01-01T00:00:10.000Z",
              "httpOnly": true,
              "name": "_zm_ctaid",
              "path": "/",
              "secure": true,
              "value": "[REDACTED]"
            },
            {
              "domain": ".zoom.us",
              "expires": "1970-01-01T00:00:10.000Z",
              "httpOnly": true,
              "name": "_zm_chtaid",
              "path": "/",
              "secure": true,
              "value": "[REDACTED]"
            }
          ],
          "headers": [
            {
              "name": "date",
              "value": "Wed, 03 Nov 2021 05:11:53 GMT"
            },
            {
              "name": "content-type",
              "value": "application/json;charset=UTF-8"
            },
            {
              "name": "content-length",
              "value": "895"
            },
            {
              "name": "connection",
              "value": "close"
            },
            {
              "name": "x-zm-trackingid",
              "value": "v=2.0;clid=aw1;rid=WEB_714c875ae41b0924ec00cfa89782a8e4"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "cache-control",
              "value": "no-cache, no-store, must-revalidate, no-transform"
            },
            {
              "name": "pragma",
              "value": "no-cache"
            },
            {
              "name": "expires",
              "value": "Thu, 01 Jan 1970 00:00:00 GMT"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "name": "vary",
              "value": "Origin, Access-Control-Request-Method, Access-Control-Request-Headers"
            },
            {
              "name": "x-ratelimit-category",
              "value": "Medium"
            },
            {
              "name": "x-zm-zoneid",
              "value": "VA2"
            },
            {
              "name": "strict-transport-security",
              "value": "max-age=31536000; includeSubDomains"
            },
            {
              "name": "x-xss-protection",
              "value": "1; mode=block"
            },
            {
              "name": "referrer-policy",
              "value": "strict-origin-when-cross-origin"
            }
          ],
          "headersSize": 1348,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2021-11-03T05:11:53.063Z",
        "time": 929,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 929
        }
      }
    ],
    "pages": [],
    "version": "1.2"
  }
}
