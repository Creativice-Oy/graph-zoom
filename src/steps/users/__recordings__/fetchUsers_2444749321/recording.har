{
  "log": {
    "_recordingName": "fetchUsers",
    "creator": {
      "comment": "persister:JupiterOneIntegationFSPersister",
      "name": "Polly.JS",
      "version": "4.3.0"
    },
    "entries": [
      {
        "_id": "2c0dedea0d862606822a0b84661af260",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "*/*"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip,deflate"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "api.zoom.us"
            }
          ],
          "headersSize": 887,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [
            {
              "name": "page_size",
              "value": "10"
            },
            {
              "name": "page_number",
              "value": "1"
            }
          ],
          "url": "https://api.zoom.us/v2/users?page_size=10&page_number=1"
        },
        "response": {
          "bodySize": 515,
          "content": {
            "mimeType": "application/json;charset=UTF-8",
            "size": 515,
            "text": "{\"page_count\":1,\"page_number\":1,\"page_size\":10,\"total_records\":1,\"users\":[{\"id\":\"Qky6bkpAS1KoDGWeh6dsMg\",\"first_name\":\"Paul Grant\",\"last_name\":\"Ilaga\",\"email\":\"pgilaga@gmail.com\",\"type\":1,\"pmi\":6629177735,\"timezone\":\"\",\"verified\":1,\"created_at\":\"2019-01-23T11:23:52Z\",\"last_login_time\":\"2021-10-24T05:51:26Z\",\"last_client_version\":\"5.7.1.543(win)\",\"pic_url\":\"https://lh3.googleusercontent.com/a-/AOh14GiueuReYwcg_vJoqRWnBC8cq7GbiWmi9bdTbs55MQ\",\"language\":\"en-US\",\"phone_number\":\"\",\"status\":\"active\",\"role_id\":\"0\"}]}"
          },
          "cookies": [
            {
              "domain": ".zoom.us",
              "expires": "1970-01-01T00:00:10.000Z",
              "httpOnly": true,
              "name": "zm_aid",
              "path": "/",
              "secure": true,
              "value": "[REDACTED]"
            },
            {
              "domain": ".zoom.us",
              "expires": "1970-01-01T00:00:10.000Z",
              "httpOnly": true,
              "name": "zm_haid",
              "path": "/",
              "secure": true,
              "value": "[REDACTED]"
            },
            {
              "domain": ".zoom.us",
              "expires": "1970-01-01T00:00:10.000Z",
              "httpOnly": true,
              "name": "zm_tmaid",
              "path": "/",
              "secure": true,
              "value": "[REDACTED]"
            },
            {
              "domain": ".zoom.us",
              "expires": "1970-01-01T00:00:10.000Z",
              "httpOnly": true,
              "name": "zm_htmaid",
              "path": "/",
              "secure": true,
              "value": "[REDACTED]"
            },
            {
              "httpOnly": true,
              "name": "cred",
              "path": "/",
              "secure": true,
              "value": "[REDACTED]"
            },
            {
              "domain": ".zoom.us",
              "expires": "1970-01-01T00:00:10.000Z",
              "httpOnly": true,
              "name": "_zm_ctaid",
              "path": "/",
              "secure": true,
              "value": "[REDACTED]"
            },
            {
              "domain": ".zoom.us",
              "expires": "1970-01-01T00:00:10.000Z",
              "httpOnly": true,
              "name": "_zm_chtaid",
              "path": "/",
              "secure": true,
              "value": "[REDACTED]"
            }
          ],
          "headers": [
            {
              "name": "date",
              "value": "Sun, 24 Oct 2021 05:51:55 GMT"
            },
            {
              "name": "content-type",
              "value": "application/json;charset=UTF-8"
            },
            {
              "name": "content-length",
              "value": "515"
            },
            {
              "name": "connection",
              "value": "close"
            },
            {
              "name": "x-zm-trackingid",
              "value": "v=2.0;clid=us04;rid=WEB_1bc613b251c668c54f4d59e9962bb56a"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "cache-control",
              "value": "no-cache, no-store, must-revalidate, no-transform"
            },
            {
              "name": "pragma",
              "value": "no-cache"
            },
            {
              "name": "expires",
              "value": "Thu, 01 Jan 1970 00:00:00 GMT"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "name": "vary",
              "value": "Origin, Access-Control-Request-Method, Access-Control-Request-Headers"
            },
            {
              "name": "x-ratelimit-category",
              "value": "Medium"
            },
            {
              "name": "x-ratelimit-limit",
              "value": "2000"
            },
            {
              "name": "x-ratelimit-remaining",
              "value": "1981"
            },
            {
              "name": "x-zm-zoneid",
              "value": "VA"
            },
            {
              "name": "strict-transport-security",
              "value": "max-age=31536000; includeSubDomains"
            },
            {
              "name": "x-xss-protection",
              "value": "1; mode=block"
            },
            {
              "name": "referrer-policy",
              "value": "strict-origin-when-cross-origin"
            }
          ],
          "headersSize": 1402,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2021-10-24T05:51:54.021Z",
        "time": 1155,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 1155
        }
      }
    ],
    "pages": [],
    "version": "1.2"
  }
}
