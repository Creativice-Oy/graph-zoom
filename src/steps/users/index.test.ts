import {
  createMockStepExecutionContext,
  Recording,
} from '@jupiterone/integration-sdk-testing';
import { IntegrationConfig } from '../../config';
import { fetchUsers } from '.';
import { integrationConfig } from '../../../test/config';
import { setupZoomRecording } from '../../../test/recording';

describe('#fetchUsers', () => {
  let recording: Recording;

  beforeEach(() => {
    recording = setupZoomRecording({
      directory: __dirname,
      name: 'fetchUsers',
    });
  });

  afterEach(async () => {
    await recording.stop();
  });

  test('should collect data', async () => {
    const context = createMockStepExecutionContext<IntegrationConfig>({
      instanceConfig: integrationConfig,
    });

    await fetchUsers(context);

    expect({
      numCollectedEntities: context.jobState.collectedEntities.length,
      collectedEntities: context.jobState.collectedEntities,
      encounteredTypes: context.jobState.encounteredTypes,
    }).toMatchSnapshot();

    const users = context.jobState.collectedEntities.filter((e) =>
      e._class.includes('User'),
    );
    expect(users.length).toBeGreaterThan(0);
    expect(users).toMatchGraphObjectSchema({
      _class: ['User'],
      schema: {
        additionalProperties: false,
        properties: {
          _rawData: {
            type: 'array',
            items: { type: 'object' },
          },
          _type: { const: 'zoom_user' },
          name: { type: 'string' },
        },
      },
    });
  });
});
