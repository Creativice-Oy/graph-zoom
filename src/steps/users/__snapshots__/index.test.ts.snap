// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#fetchUsers should collect data 1`] = `
Object {
  "collectedEntities": Array [
    Object {
      "_class": Array [
        "User",
      ],
      "_key": "zoom_user:Qky6bkpAS1KoDGWeh6dsMg",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "created_at": "2019-01-23T11:23:52Z",
            "email": "pgilaga@gmail.com",
            "first_name": "Paul Grant",
            "id": "Qky6bkpAS1KoDGWeh6dsMg",
            "language": "en-US",
            "last_client_version": "5.7.1.543(win)",
            "last_login_time": "2021-10-24T05:51:26Z",
            "last_name": "Ilaga",
            "phone_number": "",
            "pic_url": "https://lh3.googleusercontent.com/a-/AOh14GiueuReYwcg_vJoqRWnBC8cq7GbiWmi9bdTbs55MQ",
            "pmi": 6629177735,
            "role_id": "0",
            "status": "active",
            "timezone": "",
            "type": 1,
            "verified": 1,
          },
        },
      ],
      "_type": "zoom_user",
      "active": true,
      "createdOn": undefined,
      "displayName": "Paul Grant Ilaga",
      "email": "pgilaga@gmail.com",
      "id": "Qky6bkpAS1KoDGWeh6dsMg",
      "name": "Paul Grant Ilaga",
      "status": "active",
      "username": "pgilaga@gmail.com",
    },
  ],
  "encounteredTypes": Array [
    "zoom_user",
  ],
  "numCollectedEntities": 1,
}
`;
